contrib_bots/lib/github_issues.py

```diff
@@ -0,0 +1,119 @@
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        # gets token from dev-secrets
+        # CONFIGURE TOKEN ON LOCAL MACHINE FOR NOW AT .zuliprc-local
+        config = six.moves.configparser.ConfigParser()
+        config.read([os.path.expanduser('~/.github-issue-bot/github-issue-bot.conf')])
+
+        REPO_NAME = config.get('github', 'github_repo')
+        REPO_OWNER = config.get('github', 'github_repo_owner')
+
+        new_content = original_content.replace('@issue', 'by {}:'.format(original_sender,))
+        # gets the repo url
+        url = 'https://api.github.com/repos/{}/{}/issues'.format(REPO_OWNER, REPO_NAME)
+
+        # signs into github using the provided username and password
+        session = github.auth()
+
+        # Gets rid of the @issue in the issue title
+        issue_title = message['content'].replace('@issue', '').strip()
+        issue_content = ''
+        new_issue_title = ''
+        for part_of_title in issue_title.split():
+            if len(new_issue_title) < 70:
+                new_issue_title += '{} '.format(part_of_title)
+            else:
+                issue_content += '{} '.format(part_of_title)
+
+        new_issue_title = new_issue_title.strip()
+        issue_content = issue_content.strip()
+        new_issue_title += '...'
+
+        # Creates the issue json, that is transmitted to the github api servers
+        issue = {
+                 'title': new_issue_title,
+                 'body': '{} **Sent by [{}](https://chat.zulip.org/#) from zulip**'.format(issue_content, original_sender),
+                 'assignee': '',
+                 'milestone': 'none',
+                 'labels': [''],
+                }
+        # Sends the HTTP post request
+        r = session.post(url, json.dumps(issue))
+
+        if r.status_code == 201:
+            # sends the message onto the 'issues' stream so it can be seen by zulip users
+            client.send_message(dict(
+                type='stream',
+                to='issues',
+                subject=message['sender_email'],
+                # Adds a check mark so that the user can verify if it has been sent
+                content='{} :heavy_check_mark:'.format(new_content),
+            ))
+        else:
+            # This means that the issue has not been sent
+            # sends the message onto the 'issues' stream so it can be seen by zulip users
+            client.send_message(dict(
+                type='stream',
+                to='issues',
+                subject=message['sender_email'],
+                # Adds a cross so that the user can see that it has failed, and provides a link to a
+                # google search that can (hopefully) direct them to the error
+                content='{} :x: Code: [{}](https://www.google.com/search?q=Github HTTP {} Error)'
+                .format(new_content, r.status_code, r.status_code),
```

Comment:
The actual text returned in the response could be also useful here. Also, I think his line should be indented.